src/retrier.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * @param {Mixed} value
+ * @returns {Boolean}
+ */
+const isArray = value => Array.isArray(value)
+
+/**
+ *
+ * @param {Mixed} value
+ */
+const isNumber = value => typeof value === 'number'
+
+/**
+ * @param {Number} m
+ * @param {Number} b
+ */
+const createLinear = function (m = 1, b = 0) {
```

Comment:
I feel like `createLinear` should really just be a special case of `createExponential` (with `c` = 1, and taking adding the `b` option to `createExponential` (which would also be useful for some curves there anyway)).

Reply:
Also, are these `m`, `b`, `c` options standard mathematical notation?  They're far from obvious to me :D  Can better names be picked and/or better docs for them added?

Reply:
Also what do you think about using object destructuring to allow more flexible configuration of options, e.g. if the signature was like this

```
const createLinear = function ({ m = 1, b = 0 } = {}) {
```

then it would be possible to specify `b` without also having to specify `m`.

Reply:
So... `m` and `b` is the standard naming convention used for those particular constants in a "line" in cartesian geometry, yes. But that is not true of `c` and `m` in an exponential function... those should likely be change to `a` and `b` (and we could use `y = (a * b) ** x`) instead of the what I've done.... which just makes strigin writing exponential backoff a little easier to understand.

And exponential function can't adequately be decribed using a linear function, and visa versa. In the equation for a straight line (`y = (m * x) + b`) `m` is the gradient, of the line, or how much it travels along the `y` axis relative to movement on the `x` axis, and `b` is the point at which that line crosses the `x` axis. An exponential curve might never (or at least in practive never) cross either axis.

Reply:
> Also what do you think about using object destructuring to allow more flexible configuration of options

That makes sense to me.

Reply:
I'm not convinced.

```
// here are your standard linear and exponentials then
const lin = (m, x, b) => m * x + b;                                             
const exp = (c, x, m) => (c ** x) * m;                                          

// So if we allow exp to take in a `b`
const exp2 = (c, x, m, b) => (c ** x) * m + b;
// then lin can be implemented using it
const lin2 = (m, x, b) => exp2(m, 1, x, b);
```                                          

which makes sense, because if `x` is always 1 in `(c ** x)`, then `(c ** x) === c)` and so `(c ** x) * m + b` becomes `c * m + b`... which is your linear function.